### PR TITLE
Add symlink creation to framework

### PR DESCRIPTION
**Pull request by hgy59**
_Motivation:_  This PR was initiated by a question by @ymartin59 
_Linked issues:_  https://github.com/SynoCommunity/spksrc/pull/3780#discussion_r346526543

There is no automatic symlink creation as assumed in the related comment.
It turned out that there are always only some particular binaries to add to the path.

This solution handles two new Makefile variables:

`SPK_COMMANDS` is used for simple links from `${SYNOPKG_PKGDEST}` to `/usr/local`. The parent folder (`bin` or `sbin`) must be provided since the solution is implemented independent.

`SPK_LINKS` is provided for links with different source and targets. This supports a different target or to rename the command. The links are defined as <full_target>:<pkg_relative_source>

***Examples:***
`SPK_COMMANDS = bin/git bin/git-cvsserver bin/git-receive-pack bin/git-shell bin/git-upload-archive bin/git-upload-pack`
`SPK_LINKS = /usr/local/bin/tmux:bin/tmux-utf8`

### Checklist
- [x] Build rule `all-supported` completed successfully
- [x] Package upgrade completed successfully
- [x] New installation of package completed successfully

### Remarks
- most of the related packages are not updated now (most of the cli tools don't have data to migrate anyway)
- synocli-* have updated `SPK_REV`

### Related packages
Updated and tested packages are all of `/diyspk` plus:
- beets
- borgbackup
- chromaprint
- comskip
- git
- he853
- itools
- lua
- mediainfo
- rsnapshot
- synocli-disk
- synocli-file
- synocli-net
- tcl
- tmux
- zsh
- zsh-static

### Packages to update
- synocli-file due **rmlint fix** and **mc links**
- synocli-disk (description for archs not supporting DAR fixed)
- synocli-net (optional, description for ppc853x fixed)

### CODE REVIEW

**Comment by ymartin59** [comment]
@hgy59 Marvelous. I had the very same idea. I wonder why this PR contains changed 146 files. I will review.

**Comment by chickenandpork** [comment]
:) awesome.  Thank you!

**Comment by chickenandpork** [comment]
If it’s too many changed files — perhaps clouding one-change-one-PR — make the change with one package edited to use, then PR the rest of the changes later?

**Comment by hgy59** [comment]
I cleand up all the PLIST files (that is a lot)
Regarding the symlink creation I propose to rework all in one, even most of the packages are not updated now. Since this is a framework change, it is better to remove all the old versions as any package could be used as a template for new packages.
Further more it is difficult for contributers to know/remember all the changed concepts and incorporate such while updating existing or creating new packages.
(e.g. there are still packages that are not updated to the DSM6 service and user concept - and we are expecting DSM7 this year!).

There are still CLI packages that do not create symlinks yet (or do not compile, or I left for later).
These are for future PRs:
- mono
- openssh
- rsnapshot
- gnupg
- htop
- imagem0agick
- irssi
- jupp
- links
- mercurial
- mutt
- plowshare
- rdiff-backup
- sshfs
- vim

and maybe more... will be another 30+ files changed...

File: spk/ddrescue/Makefile [hunk 1]
```diff
@@ -1,26 +1,27 @@
...
-DESCRIPTION = GNU ddrescue is a data recovery tool. It copies data from one file or block device i.e. hard disc, cdrom, etc to another, trying to rescue the good parts first in case of read errors
+DESCRIPTION = GNU ddrescue is a data recovery tool. It copies data from one file or block device \(hard disc, cdrom, etc\) to another, trying to rescue the good parts first in case of read errors.
+DESCRIPTION_FRE = GNU ddrescue est un outil de récupération de données. Il copie les données provenant d\'un fichier ou d\'un périphérique mode bloc \(disque dur, cdrom, etc\) vers un autre, en essayant toutes les astuces possibles afin de récupérer des données en cas d\'erreurs de lectures.
+DESCRIPTION_ITA = GNU ddrescue è uno strumento di recupero dati. Copia i dati da un file o dispositivo a blocchi \(hard disk, cdrom, ecc.\) ad un altro, cercando fortemente di recuperare i dati in caso di errori di lettura.
+DESCRIPTION_SPN = GNU ddrescue es una herramienta de recuperación de datos. Copia los datos de un fichero o un dispositivo de bloques \(disco duro, cdrom, etc\) a otro, intentando recuperar primero las partes no dañadas en caso de producirse errores de lectura.
```
[the remark this answers is not in the document]
Comment (hgy59) [reply spk/ddrescue/Makefile]: Just took it from the original [homepage](https://www.gnu.org/software/ddrescue/ddrescue.html)

File: mk/spksrc.service.create_links [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        fi
+    done
+
+    for item in ${SPK_LINKS}
+    do
+        linkArray=(${item//:/ })
+        _link=${linkArray[0]}
+        _target=${linkArray[1]}
+        if [ -e "${SYNOPKG_PKGDEST}/${_target}" ]; then
+            mkdir -p "${_link%/*}"
```
[the remark this answers is not in the document]
Comment (hgy59) [reply mk/spksrc.service.create_links]: thanks for the hint ... I have to redesign this

File: spk/synocli-disk/Makefile [hunk 3]
```diff
@@ -1,29 +1,43 @@
...
 SPK_VERS = 1.0
-SPK_REV = 1
+SPK_REV = 2
 SPK_ICON = src/synocli-disk.png
 
-MAINTAINER = SynoCommunity
-
 include ../../mk/spksrc.common.mk
 
 DEPENDS = cross/e2fsprogs cross/fuse cross/testdisk cross/ncdu cross/davfs2 cross/lsscsi
```
[the remark this answers is not in the document]
Comment (hgy59) [reply spk/synocli-disk/Makefile]: your are right!

File: cross/ddrescue/Makefile [hunk 4]
```diff
@@ -4,12 +4,13 @@ PKG_EXT = tar.lz
...
-SRC_DIR= $(WORK_DIR)/$(PKG_NAME)-$(PKG_VERS)/
 
 DEPENDS =
 
 HOMEPAGE = http://www.gnu.org/software/ddrescue/
 COMMENT = GNU ddrescue is a data recovery tool. It copies data from one file or block device (hard disc, cdrom, etc) to another, trying hard to rescue data in case of read errors.
 LICENSE = GNU GPL
 
+ADDITIONAL_CXXFLAGS = -Wall -W -O2
+
```
[the remark this answers is not in the document]
Comment (hgy59) [reply cross/ddrescue/Makefile]: Just added the CXXFLAGS that are (unintentially) removed by the patch.
As these are related to display of warnings and code optimization the risk to break something is very low.
I admit to have tested only the working symlinks (by calling each tool with --version or --help) on x64 and armada370.
So I will dig into a real usage of ddrescue (with a small file ;-)

[the remark this answers is not in the document]
Comment (hgy59) [reply cross/ddrescue/Makefile]: Works well on armada370 and x64. 
Just "recovered" a 16GB USB medium on both.

File: cross/tcl/Makefile [hunk 5]
```diff
@@ -1,5 +1,6 @@
 PKG_NAME = tcl
 PKG_VERS = 8.6.10
+MAIN_VERS = 8.6
```
[the remark this answers is not in the document]
Comment (hgy59) [reply cross/tcl/Makefile]: I prefer **PKG_MAIN_VERS** = 8.6
regarding semantic versioning "MAJOR" has a different meaning.
MAJOR_VERS = 8
MINOR_VERS = 6
PATCH_VERS = 10

File: mk/spksrc.service.create_links (hunk 2 above)
[the remark this answers is not in the document]
Comment (hgy59) [reply mk/spksrc.service.create_links]: solved

File: spk/synocli-disk/Makefile (hunk 3 above)
[the remark this answers is not in the document]
Comment (hgy59) [reply spk/synocli-disk/Makefile]: done

File: cross/tcl/Makefile (hunk 5 above)
[the remark this answers is not in the document]
Comment (ymartin59) [reply cross/tcl/Makefile]: OK with that. Idea is to avoid two distinct variables, with the risk to forget one at upgrade.

**Pull request merged**